### DISCUSSION TITLE
Does channel deploy script need a return?

### DISCUSSION THREAD

**Discussion by tom08zehn** [nextgenhealthcare/connect#4929] [Q&A] [answered]
In many examples I see that the channel deploy script ends with a `return;` statement.

Question 1: If it returns nothing, do I need a `return` at all _at the end of the script_?
Question 2: What happens if I do `return myObject;` (return _something_, whatever it may contain...)?

(clear: if I want to interrupt/break out of the script somewhere in the middle then I can do it like this `if(this) return;`)

**Comment by tonygermano** [comment, marked as answer]
Returning early will short circuit your script, but it won't prevent the channel from deploying.

As far as I can tell from looking through the source, mirth does not inspect the return value, so you could probably return anything you wanted. `return;` is the same as `return undefined;`, and there is an implicit `return undefined;` at the end of every javascript function. If you actually want to stop the channel from deploying, throw something.

**Reply by tom08zehn** [reply]
Many thanks.

I don't want to prevent the channel from deploying. I just want to omit the `return;` if not needed and thereby wanted to understand Mirth's behavior.
